server/src/lib/api.ts

```diff
@@ -133,9 +132,11 @@ export default class API {
           sentences = sentences.concat.apply(sentences, sentenceArrays);
           console.log('sentences found', sentences.length);
           this.sentencesCache = sentences;
+          return this.sentencesCache;
         })
         .catch((err: any) => {
           console.error('could not retrieve sentences', err);
+          return [];
```

Comment:
we now get better typechecking (`bluebird` was used without typings). this method actually returned a `Promise<void>` when the "actually do work" case happened.

I guess the consuming code dealt with this in some way, else we would have noticed problems. but now it is actually correct :)

note: returning the empty array in from the catch clause could be replaced by returning nothing and widening the type signature to `Promise<string[] | void>` (the consuming code then needs to check for `undefined`, but that's probably already the case)